gcsio/src/test/java/com/google/cloud/hadoop/gcsio/TrackingHttpRequestInitializer.java

```diff
@@ -169,12 +171,33 @@ public ImmutableList<String> getAllRequestStrings() {
         .collect(toImmutableList());
   }
 
+  public ImmutableList<String> getAllRequestInvocationIds() {
```

Comment:
where is this getting used?

Reply:
We should try to avoid adding code just for enabling testing. Aren't there other alternatives?

Reply:
I am confused.
This is the code in `test` directory. 
It is serving the purpose of extracting specific information from Http request header.
It is a required piece of code.
____
What is your proposal to `alternatives`?

Reply:
My bad. I misread.